src/Storages/StorageReplicatedMergeTree.cpp

```diff
@@ -1411,9 +1415,79 @@ void StorageReplicatedMergeTree::drop()
 }
 
 
+/// Removes zero-copy locks and their parent nodes if there are no more children
+static bool tryRemoveZeroCopyLocks(zkutil::ZooKeeperPtr zookeeper, const std::string & root_path, const std::string & replica_name)
+{
+    std::function<bool(const std::string &)> remove_recursive = [&] (const std::string & path)
+    {
+        Strings children;
+        if (zookeeper->tryGetChildren(path, children) != Coordination::Error::ZOK)
```

Comment:
This method is pretty similar to `ZooKeeper::tryRemoveChildrenRecursive`, but it also allows to delete intermediate nodes for which all children nodes with lock are deleted. Another solution would be adding a new parameter to `ZooKeeper::tryRemoveChildrenRecursive`, which allows to drop children only with the given name. The only drawback is that this would leave some garbage intermediate nodes in the ZK, but maybe it's ok.